src/feedParserPromised.js

```diff
@@ -2,27 +2,85 @@
 
 const request = require('request');
 const FeedParser = require('feedparser');
+const Iconv = require('iconv').Iconv;
+const zlib = require('zlib');
+
+let items = [];
+let feedparser;
 
 module.exports = class FeedParserPromised {
   static parse (requestOptions, feedparserOptions) {
-    return new Promise( (resolve, reject) => {
-      const items = [];
-      const feedparser = new FeedParser(feedparserOptions);
+    return new Promise((resolve, reject) => {
+      requestOptions = typeof requestOptions == 'string' ? { uri: requestOptions, onError: reject } : requestOptions;
```

Comment:
Have you considered creating some unit tests for this logic? 

Suggestion: maybe it's an opportunity to extract this into a different abstraction like `RequestParser` that could be injected on this function, which would also be easier to test.